selene-lib/src/rules/roblox_incorrect_roact_usage.rs

```diff
@@ -43,10 +43,7 @@ impl Rule for IncorrectRoactUsageLint {
         for unused_property in visitor.unused_properties {
             diagnostics.push(Diagnostic::new(
                 "roblox_incorrect_roact_usage",
-                format!(
-                    "`{}` will never be applied for `{}`",
-                    unused_property.property_name, unused_property.class_name
-                ),
+                format!("`{}` will never be applied", unused_property.property_name),

```

Comment:
Like I said, let's clarify that this is for native Roblox UI only.

At the same time, we should also show them the correct thing to do when the property is `Name`, as it seems to confuse people on Discord.

Reply:
Not entirely sure how I should phrase the help message, without it being too long. Also - since we don't have access to the context surrounding it, its not possible to differentiate between the top-most component (passed to `Roact.mount(component, target, name)`), or if its a child component, where the name is inferred from the key - so it would most likely have to mention both of them, right?

Reply:
It could only mention it when it's inside a table of its own, as a secondary message saying "try: ...", like rustc.

Reply:
The visitor currently only visits `FunctionCall` for these - it doesn't know that its inside a table. If I wanted to detect if it was inside a table, wouldn't I have to add a visitor to visit a `TableConstructor`, then go through its fields and recursively check all of them for a `createElement` or equiv. FunctionCall, whilst also keeping the `FunctionCall` visitor to lint other components outside of tables?

Reply:
My approach to this is usually capturing the spans of every table, but if that's too much of a headache then it might not be worth it and can be degerred.

Reply:
That does sound like it should work, and shouldn't be too bad to implement, much easier than what I was thinking :) - I'll take a look.